### PR TITLE
[SearchInputMain] - task/searchInputMain-improvements

### PR DESCRIPTION
**Pull request by Scythe14**
- Improvements

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -2,7 +2,7 @@
 buildscript {
 
     ext {
-        grapes_version = '1.0.6'
+        grapes_version = 'develop-SNAPSHOT'
```
Comment (Tweener) [inline build.gradle:5]: We don't need to change the version number here. 
`develop-SNAPSHOT` is the name of the version targeting `develop` branch that Jitpack automatically generate. 
Here we only use a version number.

File: library/src/main/java/com/spendesk/grapes/inputs/SearchInputMain.kt [hunk 2]
```diff
@@ -52,50 +54,68 @@ class SearchInputMain : CardView {
...
 
-    lateinit var editText: AppCompatEditText
-    private var style = Style.getDefault()
-
     fun setStyle(style: Style) {
         when (style) {
             Style.PRIMARY -> {
-                radius = resources.getDimensionPixelOffset(R.dimen.searchInputMainPrimaryCardRadius).toFloat()
-                elevation = resources.getDimensionPixelOffset(R.dimen.searchInputMainPrimaryCardElevation).toFloat()
+                radius = resources.getDimensionPixelOffset(R.dimen.inputSearchMainPrimaryCardRadius).toFloat()
```
Comment (Tweener) [inline library/src/main/java/com/spendesk/grapes/inputs/SearchInputMain.kt:60]: Just out of curiosity, why did you rename from `searchInputMain...` to `inputSearchMain...` when the name of the class is `SearchInputMain`? (same goes for all the other naming in this class).
Did you also want to rename this class?

File: library/src/main/res/layout/input_search_main.xml [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+            style="@style/InputSearchMainEditText"
+            android:layout_width="0dp"
+            android:layout_height="wrap_content"
+            android:padding="16dp"
+            app:layout_constraintEnd_toEndOf="parent"
+            app:layout_constraintStart_toStartOf="parent"
+            app:layout_constraintTop_toTopOf="parent" />
+
+        <View
+            android:id="@+id/guideline"
```
Comment (Tweener) [inline library/src/main/res/layout/input_search_main.xml:24]: It might be better to rename it `inputSearchMainGuideline` in case there are other guidelines, to avoid confusion.

File: library/src/main/res/layout/input_search_main.xml [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+            app:layout_constraintBottom_toBottomOf="parent"
+            app:layout_constraintEnd_toEndOf="parent"
+            app:layout_constraintStart_toStartOf="parent" />
+
+        <ProgressBar
+            android:id="@+id/inputSearchMainProgressBar"
+            style="@style/InputSearchMainProgressBar"
+            android:layout_width="match_parent"
+            android:layout_height="wrap_content"
+            android:indeterminate="true"
```
Comment (Tweener) [inline library/src/main/res/layout/input_search_main.xml:38]: This property is already defined in the style.

**Review by Tweener — CHANGES_REQUESTED** [review]
It looks really good! Great job 🔥 

Just a few things, check my comments.

Also, I run the app on the sample, and the progress bar looks clear blue instead of purple. Can you check why the progressDrawable does not seem to be applied?
Capture d’écran 2021-07-30 à 07 54 17

File: build.gradle (hunk 1 above)
Comment (Scythe14) [reply build.gradle:5]: Yes of course, it was for my tests .. I put it back !

File: library/src/main/java/com/spendesk/grapes/inputs/SearchInputMain.kt (hunk 2 above)
Comment (Scythe14) [reply library/src/main/java/com/spendesk/grapes/inputs/SearchInputMain.kt:60]: It is again all about the hierarchy on the design system that we need to redo, prefix input because of the hierarchy, have not renamed the whole class to avoid impact on the spendesk app, but I could anyway

**Review by Tweener — APPROVED** [review]
Ouh yeah 🕺
